tests/rptest/clients/oauth_producer.py

```diff
@@ -0,0 +1,66 @@
+# Copyright 2023 Redpanda Data, Inc.
+#
+# Use of this software is governed by the Business Source License
+# included in the file licenses/BSL.md
+#
+# As of the Change Date specified in that file, in accordance with
+# the Business Source License, use of this software will be governed
+# by the Apache License, Version 2.0
+
+# TODO: Different grant types
+# TODO: Metadata reads don't seem to work (same issue as AdminClient)
```

Comment:
interesting. is this when interacting with vanilla kakfa since our oauth support isn't yet available?

Reply:
No, but I should try that. What I've observed is that metadata requests made with librdkafka + OAUTHBEARER time out without ever triggering the OIDC flow.

If it were an issue of the backend not supporting the auth type, I would expect the client to first try to obtain a bearer token, attempt the request, and fail. Not sure what's happening there, but I'll try it against kafka as well.

Reply:
nvm - looks to be by design. Only way to trigger the oauth callback is `client.poll()` or `producer.flush()`.

Reply:
ah, good to know